### PR TITLE
[Feature] Add fault injection evaluation kit

### PR DESCRIPTION
**Pull request by zhuangwang93**
## Problem

Fault injection existed only as a test helper and could not run reproducible, framework-aware campaigns or compare neutral localization results with verified ground truth.

## Implementation

- Add a dedicated `lm_resiliency/fault_injection/` package with declarative campaign manifests, framework model resolution, scheduled transient and persistent numerical faults, output delays, restoration, and verified injection records.
- Support PyTorch, TorchTitan, Megatron Core, and DeepSpeed training-object surfaces without resiliency-system-specific adapters.
- Add neutral `LocalizationResult` scoring and JSON campaign reports containing the full manifest and per-occurrence ground truth.
- Replace the duplicated test-only injector, expose the stable package-root API, and add a CPU example and user guide.
- Update the roadmap to distinguish the implemented safe in-process slice from planned destructive and state-corruption extensions.

## Compatibility

- The feature is opt-in and does not change `enable_resiliency` behavior.
- The current safety boundary is numerical parameter/output corruption and module-output delay. Process, node, communication, checkpoint, gradient, and optimizer-state faults remain planned.

## Validation

- `CUDA_VISIBLE_DEVICES="" python -m pytest -q` (664 passed)
- `ruff check .`
- `ruff format --check .`
- `pre-commit run --all-files`
- `python examples/fault_injection.py`
- `python -m build`

### CODE REVIEW

**Comment by zhuangwang93** [comment]
Closing before review at the author's request; the implementation will remain local for further design discussion.
